### ISSUE TITLE
`async` attribute not added in dev server

### ISSUE THREAD

**Issue by thdoan** [vuejs/vitepress#4272]
### Describe the bug

I followed [this example](https://vitepress.dev/reference/site-config#example-using-google-analytics) to add a GA tag, but on the dev server I do not see the `async` attribute:

![image](https://github.com/user-attachments/assets/6bdc29f1-6035-4d9e-9b31-f2e8da7aaa1b)

I'm only developing, so don't know if this is a bug or the expected behavior only on the dev server.

### Reproduction

1. Add GA tag as shown in [this example](https://vitepress.dev/reference/site-config#example-using-google-analytics)
2. Run `npm run dev` to start the dev server
3. Load your site

### Expected behavior

```
<script async src="https://www.googletagmanager.com/gtag/js?id=TAG_ID"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'TAG_ID');
</script>
```

### System Info

```Text
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 13.10 GB / 31.95 GB
  Binaries:
    Node: 18.20.4 - C:\Program Files\nodejs\node.EXE
    npm: 10.7.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.19041.4355
  npmPackages:
    vitepress: latest => 1.3.4
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by thdoan** [comment]
UPDATE: I just published and noticed the `async` attribute _is_ on the production site, so it's only missing on the dev server. Not sure if this is the intended behavior. If it is, you can close this ticket.

**Comment by brc-dd** [comment]
During dev, the scripts are added dynamically. Dynamically added scripts are async by default. You can try doing `$0.async` in devtools console.

**Comment by brc-dd** [comment]
fixed in 461a5b001d29f95169f60fe28bc610e3f6e8fd66
